app/views/admin/users/edit.html.erb

```diff
@@ -0,0 +1,82 @@
+              <%= @user.errors.full_message(:current_password, @user.errors[:current_password].first) %>
+            </div>
+          <% end %>
+        </div>
+      <% end %>
+
+      <div class="form-control w-full mt-4">
+        <%= f.label :password, class: "label-text text-black mb-2 font-semibold" %>
+        <%= f.password_field :password, autocomplete: "new-password", class: "input input-bordered w-full" + (@user.errors[:password].any? ? " input-error " : " border-black ") %>
+        <% if @minimum_password_length %>
+          <span class="label-text text-black mt-2">
+            <% if @minimum_password_length %><%= @minimum_password_length %> characters minimum.<% end %> Leave blank if you don't want to change it.</span>
+        <% end %>
+        <% if @user.errors[:password].any? %>
+          <div class="field-error label-text mt-2 text-red-700">
+            <%= @user.errors.full_message(:password, @user.errors[:password].first) %>
+          </div>
+        <% end %>
+      </div>
+
+      <div class="form-control w-full mt-4">
+        <%= f.label :password_confirmation, class: "label-text text-black mb-2 font-semibold" %>
+        <%= f.password_field :password_confirmation, autocomplete: "new-password", class: "input input-bordered w-full" + (@user.errors[:password_confirmation].any? ? " input-error " : " border-black ") %>
+        <% if @user.errors[:password_confirmation].any? %>
+          <div class="field-error label-text mt-2 text-red-700">
+            <%= @user.errors.full_message(:password_confirmation, @user.errors[:password_confirmation].first) %>
+          </div>
+        <% end %>
+      </div>
+
+      <div class="w-full mt-4">
+        <%= f.check_box :is_site_admin %>
+        <%= f.label :is_site_admin, class: "label-text text-black mb-2 font-semibold" %>
+        <div class="label-text text-black mt-2">User should be able to see and edit all sites and documents.</div>
+      </div>
+
+      <div class="w-full mt-4">
+        <%= f.check_box :is_user_admin %>
+        <%= f.label :is_user_admin, class: "label-text text-black mb-2 font-semibold" %>
+        <div class="label-text text-black mt-2">User should be able to edit other users and create admins.</div>
+      </div>
+
+      <div class="w-full mt-4">
+        <%= f.label :site_id, "Site", class: "label-text text-black mb-2 font-semibold" %>
+        <%= f.select :site_id, grouped_options_for_select(@sites, @user.site_id),
+                     { include_blank: "None" },
+                     { class: "input input-bordered w-full select" } %>
+      </div>
+
+      <div class="actions flex gap-4 items-center mt-6">
+        <%= button_tag type: "submit", class: "btn btn-primary text-white", id: "submit-user-form" do %>
+          <i class="fas fa-check mr-2"></i>
+          Update
+        <% end %>
+        <%= link_to "Back", :back %>
+      </div>
+    <% end %>
+  </div>
+</div>
```

Comment:
Getting funny behavior here. I thought I needed to supply the current password on a user to edit the checkboxes, but maybe I don't? I was logged in as the listed user, which had user admin access.

https://github.com/user-attachments/assets/41957104-215f-4b8c-8bbd-1d41539aa04d

Reply:
Yeah, maybe just "We need your current password to confirm **any password changes**"? I think this error doesn't happen if you just select / unselect the admin boxes without supplying the current password.